LeaderboardBackend/Controllers/AccountController.cs

```diff
@@ -114,4 +116,60 @@ public async Task<ActionResult<LoginResponse>> Login([FromBody] LoginRequest req
             badCredentials => Unauthorized()
         );
     }
+#pragma warning restore CS1573
+
+    /// <summary>
+    ///     Resends the account confirmation link.
+    /// </summary>
+    /// <param name="authService">IAuthService dependency.</param>
+    /// <param name="confirmationService">IUserConfirmationService dependency.</param>
+    /// <param name="emailSender">EmailSender dependency.</param>
+    /// <response code="200">The request was sent successfully.</response>
+    /// <response code="400">
+    ///     The request was malformed.
+    /// </response>
+    /// <response code="401">
+    ///     The request doesn't contain a valid session token.
+    /// </response>
+    /// <response code="409">
+    ///     A `User` with the specified username or email already exists.<br/><br/>
+    ///     Validation error codes by property:
+    ///     - **Username**:
+    ///       - **UsernameTaken**: the username is already in use
+    ///     - **Email**:
+    ///       - **EmailAlreadyUsed**: the email is already in use
+    /// </response>
+    /// <response code="500">
+    ///     Internal server error.
+    /// </response>
+    [HttpPost("confirm")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status409Conflict)]
+    [ProducesResponseType(StatusCodes.Status429TooManyRequests)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult> ResendConfirmation(
+        [FromServices] IAuthService authService,
+        [FromServices] IUserConfirmationService confirmationService,
+        [FromServices] IEmailSender emailSender
+    )
+    {
+        // TODO: Handle rate limiting (429 case) - zysim
+
+        GetUserResult result = await _userService.GetRegisteredUserFromClaims(HttpContext.User);
+
+        if (result.TryPickT0(out User user, out OneOf<BadCredentials, UserNotFound, BadRole> errors))
+        {
+            await confirmationService.CreateConfirmation(user);
+            return Ok();
+        }
+
+        return errors.Match(
+            badCredentials => Unauthorized(),
+            // Shouldn't be possible; throw 500
+            notFound => StatusCode(StatusCodes.Status500InternalServerError),
```

Comment:
Status code 500 should be reserved for actual server errors. Return 401 instead.